server/base/src/main/java/org/apache/accumulo/server/manager/state/MergeInfo.java

```diff
@@ -84,13 +84,22 @@ public boolean isDelete() {
   public boolean needsToBeChopped(KeyExtent otherExtent) {
     // During a delete, the block after the merge will be stretched to cover the deleted area.
     // Therefore, it needs to be chopped
-    if (!otherExtent.tableId().equals(extent.tableId())) {
-      return false;
-    }
-    if (isDelete()) {
+    // if (!otherExtent.tableId().equals(extent.tableId())) {
+    // return false;
+    // }
+    // if (isDelete()) {
+    // return otherExtent.prevEndRow() != null && otherExtent.prevEndRow().equals(extent.endRow());
+    // } else {
+    // return this.extent.overlaps(otherExtent);
+    // }
```

Comment:
Or rather, it would have been easier if the original code had a comment explaining what it was doing, so that the diff when it's removed would also remove the comment, and it would be obvious what case is being covered here :smiley_cat: